### ISSUE TITLE
KSQL should support multiple expressions in a JOIN clause

### ISSUE THREAD

**Issue by rmoff** [confluentinc/ksql#2344]
KSQL currently only supports a single expression in a JOIN clause. It should support multiple expressions. 

Doesn't work:

```
ksql> SELECT 'Bus: ' + CAST(EV.BUS_ID AS VARCHAR), \
>       'Route: ' + CAST(EV.ROUTE_ID AS VARCHAR), \
>       'Stop: ' + CAST(SCH.STOP_NUM AS VARCHAR), \
>       'Due: ' + TIMESTAMPTOSTRING(SCH.ETA, 'yyyy-MM-dd  HH:mm:ss Z') AS ETA, \
>       'Actual: ' + TIMESTAMPTOSTRING(EV.TIMESTAMP, 'HH:mm:ss Z') AS ACTUAL_TIME, \
>       'Difference (minutes): ' + CAST((EV.TIMESTAMP-SCH.ETA)/1000/60 AS VARCHAR) AS MINUTES_DELAYED \
>  FROM BUS_EVENTS EV \
>       LEFT OUTER JOIN \
>         BUS_SCHEDULE SCH \
>         ON EV.LAST_STOP = SCH.STOP_NUM AND  EV.ROUTE_ID = SCH.ROUTE_ID;
io.confluent.ksql.parser.tree.LogicalBinaryExpression cannot be cast to io.confluent.ksql.parser.tree.ComparisonExpression
```

Clumsy workaround: 

```
ksql> SELECT 'Bus: ' + CAST(EV.BUS_ID AS VARCHAR), \
>       'Route: ' + CAST(EV.ROUTE_ID AS VARCHAR), \
>       'Stop: ' + CAST(SCH.STOP_NUM AS VARCHAR), \
>       'Due: ' + TIMESTAMPTOSTRING(SCH.ETA, 'yyyy-MM-dd  HH:mm:ss Z') AS ETA, \
>       'Actual: ' + TIMESTAMPTOSTRING(EV.TIMESTAMP, 'HH:mm:ss Z') AS ACTUAL_TIME, \
>       'Difference (minutes): ' + CAST((EV.TIMESTAMP-SCH.ETA)/1000/60 AS VARCHAR) AS MINUTES_DELAYED \
>  FROM BUS_EVENTS EV \
>       LEFT OUTER JOIN \
>         BUS_SCHEDULE SCH \
>         ON EV.LAST_STOP = SCH.STOP_NUM \
>WHERE  EV.ROUTE_ID = SCH.ROUTE_ID;
```

(repo to reproduce above SQL : https://github.com/confluentinc/ksql-recipes-try-it-at-home/tree/master/calculating-bus-delay-time)
